lib/simplifyExpression/substituteScopeSearch/index.js

```diff
@@ -0,0 +1,54 @@
+const ChangeTypes = require('../../ChangeTypes');
+const Node = require('../../node');
+const TreeSearch = require('../../TreeSearch');
+
+// Searches through the tree, prioritizing deeper nodes, and substitutes
+// in-scope values for their respective expressions on a symbol node if possible.
+// Returns a Node.Status object.
+const search = TreeSearch.postOrder(scopeSubstitution);
+
+function scopeSubstitution(node, scope) {
+  if (Node.Type.isSymbol(node)) {
+    return substituteAndSimplifySymbols(node, scope);
+  }
+  else {
+    return Node.Status.noChange(node);
+  }
+}
+
+// SUBSTITUTES
+// Returns a Node.Status object with substeps
+function substituteAndSimplifySymbols(node, scope) {
+  if (!Node.Type.isSymbol(node)) {
+    return Node.Status.noChange(node);
+  }
+
+  let symbolName;
+  if (node.type === 'SymbolNode') {
+    symbolName = node.name;
+  }
+  else if (Node.Type.isUnaryMinus(node)) {
+    symbolName = node.args[0].name;
+  }
+
+  if (scope.hasOwnProperty(symbolName)) {
+    // when declared at top, kept getting TypeError: ___ is not a function
+    const simplifyExpression = require('../../simplifyExpression');
+    const substeps = simplifyExpression(scope[symbolName]);
```

Comment:
I'm a bit turned around here, perhaps due to the approach I took. Since simplifyExpression is where the scope substitution takes place, I'm concerned calling math.parse(scope[symbolName]) here won't be effective since it will not perform the scope substitution. That may be why you advised to put the substituteScope call deeper in the simplify function?

Left as is in my recent (incremental) PR.